### PR TITLE
Add split function to rules engine

### PR DESCRIPTION
**Pull request by mtdowling**
#### Background
* What do these changes do? 
* Why are they important?

#### Testing
* How did you test these changes?

#### Links
* Links to additional context, if necessary
* Issue #, if applicable (see [here](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword) for a list of keywords to use for linking issues)

---
By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: docs/source-2.0/additional-specs/rules-engine/standard-library.rst [hunk 1]
```diff
@@ -376,6 +376,109 @@ The following table shows valid and invalid values for an input to the
...
+Return type
+    ``array<string>``
+Since
+    1.1
+
+The ``split`` function divides a string into an array of substrings based on a non-empty delimiter.
+The behavior is controlled by the limit parameter:
+
+* ``limit = 0``: Split all occurrences (unlimited)
+* ``limit = 1``: No split performed (returns original string as single element array)
```
Comment (JordonPhillips) [inline docs/source-2.0/additional-specs/rules-engine/standard-library.rst:399]: Is this actually useful?

Comment (mtdowling) [reply docs/source-2.0/additional-specs/rules-engine/standard-library.rst:399]: It matches how Java split works. I didn't know what else to do with 1.

```
"hello".split("l", 1)[0]; // "hello"
```

**Review by kstich — APPROVED** [review]

File: smithy-rules-engine/src/test/resources/software/amazon/smithy/rulesengine/language/errorfiles/valid/split.smithy [hunk 2]
```diff
@@ -0,0 +1,869 @@
...
+                    "fn": "stringEquals",
+                    "argv": ["{Limit}", "0"]
+                },
+                {
+                    "fn": "split",
+                    "argv": ["{Input}", "{Delimiter}", 0],
+                    "assign": "parts"
+                },
+                {
+                    "fn": "coalesce",
```
Comment (alextwoods) [inline smithy-rules-engine/src/test/resources/software/amazon/smithy/rulesengine/language/errorfiles/valid/split.smithy:42]: Nice use of coalesce!

File: smithy-rules-engine/src/test/resources/software/amazon/smithy/rulesengine/language/errorfiles/valid/split.smithy [hunk 3]
```diff
@@ -0,0 +1,869 @@
...
+                    "properties": {
+                        "splitResult": "p1=a; p2=b; p3=c; p4=d; p5=e"
+                    }
+                }
+            }
+        },
+
+        // Edge cases
+        {
+            "documentation": "single character delimiter",
```
Comment (alextwoods) [inline smithy-rules-engine/src/test/resources/software/amazon/smithy/rulesengine/language/errorfiles/valid/split.smithy]: not sure I'd consider a single character delimiter as an edge case?

**Review by alextwoods — APPROVED** [review]

File: smithy-rules-engine/src/test/resources/software/amazon/smithy/rulesengine/language/errorfiles/valid/split.smithy (hunk 3 above)
Comment (mtdowling) [reply smithy-rules-engine/src/test/resources/software/amazon/smithy/rulesengine/language/errorfiles/valid/split.smithy]: Agreed, and I'm removing this test since it's well covered by other stuff
